examples/ex1_o1.cpp

```diff
@@ -0,0 +1,273 @@
+//                                MFEM Example 1
+//
+// Compile with: make ex1
+//
+// Sample runs:  ex1 -m ../data/square-disc.mesh
+//               ex1 -m ../data/star.mesh
+//               ex1 -m ../data/star-mixed.mesh
+//               ex1 -m ../data/escher.mesh
+//               ex1 -m ../data/fichera.mesh
```

Comment:
It will be nice if `ex1 -m ../data/inline-pyramid.mesh` worked. (and I'm sure someone will try it and report it)

I know we can do

```c++
      if (order == 1)
      {
         fec = new LinearFECollection();
      }
      else
      {
         fec = new H1_FECollection(order, dim);
      }
```

Is there a cleaner way to make it happen?

Reply:
The `H1_FECollection` can use the linear elements for pyramids if `p=1` and error out if asked for the pyramid element and `p` was ` > 1`. Similar thing can be done for the other arbitrary order collections too.

Reply:
That way most examples should just work with pyramids at least for the lowest order spaces.

Reply:
I can implement this if there's consensus. @tzanio?

Reply:
I like that

Reply:
Done

Reply:
I can make similar changes to `ND_FECollection`, `RT_FECollection`, and `L2_FECollection` if people are happy with this.

Reply:
This looks good to me. Lets update the other collections similarly.

We can also maybe add sample runs in some of the examples with pyramid meshes.